utils/python/CIME/system_test.py

```diff
@@ -374,19 +369,48 @@ def _create_newcase_phase(self, test):
     ###########################################################################
     def _xml_phase(self, test):
     ###########################################################################
+        test_case = CIME.utils.parse_test_name(test)[0]
 
-        # modify the CASEROOT xml files depending on the settings of case_opts from
-        # then call case.flush()
+        # Create, fill and write an envtest object
+        test_dir = self._get_test_dir(test)
+        envtest = EnvTest(test_dir)
 
         # Determine list of component classes that this coupler/driver knows how
         # to deal with. This list follows the same order as compset longnames follow.
         files = Files()
         drv_config_file = files.get_value("CONFIG_DRV_FILE")
         drv_comp = Component(drv_config_file)
         component_classes = drv_comp.get_valid_model_components()
+        envtest.add_elements_by_group(drv_comp, {}, "env_test.xml")
```

Comment:
If we could encapsulate this in Case, then you wouldn't need that flushall flag.